app/models/concerns/course/levels_concern.rb

```diff
@@ -17,28 +17,25 @@ def numbered_levels
   # the level that a course participant would have attained if
   # s/he had experience_points number of experience points.
   #
-  # The motivation for this method is to accommodate named levels.
-  #
-  # nil is returned if experience_points is insufficient to
-  # attain to any level, i.e. nil represents the zeroth level.
+  # If experience_points <= 0, the level is assumed to be the
+  # default level (the 0th level) with 0 experience_points threshold.
   #
   # @param [Fixnum] experience_points Number of Experience Points
-  # @return [Course::Level, nil] A Course::Level instance, or nil.
-  def compute_level(experience_points)
+  # @return [Course::Level] A Course::Level instance.
+  def current_level(experience_points)
     i = numbered_levels.rindex do |l|
       l.experience_points_threshold <= experience_points
     end
-    i ? numbered_levels[i] : nil
+    i ? numbered_levels[i] : numbered_levels.first
   end
 
-  # Returns the level number that a course participant would
-  # have attained if s/he had experience_points number of
-  # experience points.
-  #
-  # @param [Fixnum] experience_points Number of experience points
-  # @return [Fixnum] The level number for the given number of experience points
-  def compute_level_number(experience_points)
-    level = compute_level(experience_points)
-    level ? level.level_number : 0
+  # Test if the course has a default level.
+  # @return [Boolean] True if there is a default level, otherwise false.
+  def has_default_level?
+    if new_record?
+      levels.find(&:experience_points_threshold).present?
```

Comment:
Okay sorry I'm coming up with these weird queries because I'm still not too sure about ActiveRecord (Sorry this is probably another noob question >_<).

The problem I face is that if when the course is new, we will build the default level, and using `levels.find(experience_points_threshold: 0)` returns the id, which is nil. If the record is absent, a `ActiveRecord::RecordNotFound` error is thrown. So one way is to catch error and return false. 

The other way is to covert it into an array and search. `levels.to_ary.map(&:experience_points_threshold).include?(0)`

Or am I understanding AR wrongly?